android/src/main/java/com/alinz/parkerdan/shareextension/ShareModule.java

```diff
@@ -53,23 +68,23 @@ public WritableMap processIntent() {
         if (type == null) {
           type = "";
         }
+
         if (Intent.ACTION_SEND.equals(action) && "text/plain".equals(type)) {
-          value = intent.getStringExtra(Intent.EXTRA_TEXT);
-        }
-        else if (Intent.ACTION_SEND.equals(action) && ("image/*".equals(type) || "image/jpeg".equals(type) || "image/png".equals(type) || "image/jpg".equals(type) ) ) {
+          text = intent.getStringExtra(Intent.EXTRA_TEXT);
+        } else if (Intent.ACTION_SEND.equals(action) && type.startsWith("image")) {
           Uri uri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
-         value = "file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri);
-
-       } else {
-         value = "";
-       }
-      } else {
-        value = "";
-        type = "";
-      }
+          images.pushString("file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri));
+        } else if (Intent.ACTION_SEND_MULTIPLE.equals(action) && type.startsWith("image")) {
```

Comment:
I agree with you @npomfret. I think it would be better to return an array of {type, value}.
Also, it would be nice to be able to share other types as audio, video, etc.
I'll try to make a PR asap